packages/rln/src/contract/rln_contract.ts

```diff
@@ -388,8 +467,32 @@ export class RLNContract {
         }
       };
     } catch (error) {
-      log.error(`Error in registerWithIdentity: ${(error as Error).message}`);
-      return undefined;
+      if (error instanceof Error) {
```

Comment:
this changes the flow, before `registerWithIdentity` would return `undefined` in case of error and now it would throw

also this blob of handlers doesn't seem that useful - how do you use it outside?

Reply:
This gives more specific error messages instead of just exiting out with a `return undefined`

The usage of this is by wrapping this function call in a try-catch, and thus `catching` the wrong error message and rendering it on the UI